### ISSUE TITLE
bug(tests): docgen.parse

### ISSUE THREAD

**Issue by muescha** [gatsbyjs/gatsby#24487]
## Description

i have 2 tests failing.

### Steps to reproduce

```shell
yarn test
```

_note: since i get errors when installing with `yarn.lock` i delete it always before `yarn run bootstrap`_

### Expected result

no error

### Actual result

```shell
Summary of all failing tests
 FAIL  packages/gatsby-transformer-react-docgen/src/__tests__/displayname-handler.js
  ● DisplayNameHandler › Explicitly set displayName as static class member

    No suitable component definition found.

      14 |     ${source}
      15 |   `
    > 16 |   return docgen.parse(code, findAllComponentDefinitions, [handler])[0]
         |                 ^
      17 | }
      18 | 
      19 | describe(`DisplayNameHandler`, () => {

      at parse (node_modules/react-docgen/dist/parse.js:66:13)
      at Object.parse (node_modules/react-docgen/dist/main.js:61:29)
      at parse (packages/gatsby-transformer-react-docgen/src/__tests__/displayname-handler.js:16:17)
      at Object.<anonymous> (packages/gatsby-transformer-react-docgen/src/__tests__/displayname-handler.js:32:17)

  ● DisplayNameHandler › Infer displayName from class declaration/expression name

    No suitable component definition found.

      14 |     ${source}
      15 |   `
    > 16 |   return docgen.parse(code, findAllComponentDefinitions, [handler])[0]
         |                 ^
      17 | }
      18 | 
      19 | describe(`DisplayNameHandler`, () => {

      at parse (node_modules/react-docgen/dist/parse.js:66:13)
      at Object.parse (node_modules/react-docgen/dist/main.js:61:29)
      at parse (packages/gatsby-transformer-react-docgen/src/__tests__/displayname-handler.js:16:17)
      at Object.<anonymous> (packages/gatsby-transformer-react-docgen/src/__tests__/displayname-handler.js:67:19)
```

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here:

```shell

  System:
    OS: macOS 10.15.3
    CPU: (8) x64 Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.13.1 - ~/.nvm/versions/node/v12.13.1/bin/node
    Yarn: 1.21.0 - /usr/local/bin/yarn
    npm: 6.14.5 - ~/.nvm/versions/node/v12.13.1/bin/npm
  Languages:
    Python: 3.7.6 - /usr/local/opt/python/libexec/bin/python
  Browsers:
    Chrome: 83.0.4103.61
    Safari: 13.0.5
  npmGlobalPackages:
    gatsby-cli: 2.12.29
```

**Comment by madalynrose** [comment]
Hi there, @muescha! Could you add some more detail about how to reproduce the test failures you're seeing? Which directory are you in?

**Comment by muescha** [comment]
i am in main directory and running add the tests with `yarn test`


(Note: I do `rm -rf node_modules/ yarn.lock` and then `yarn run bootstrap` after I fetch the master because otherwise I never get some of the node sharp things installed)

**Comment by vladar** [comment]
It happens because `yarn.lock` has `react-docgen` version pinned to `5.0.0-beta.1`. And they had a breaking change in `5.0.0` which changed the way it detects react classes https://github.com/reactjs/react-docgen/pull/397

I would say to fix this we should update react-dockgen dependency to `^5.0.0` and fix tests. Tests that fail do not extend `React.Component`, e.g.:

https://github.com/gatsbyjs/gatsby/blob/317aa656e6a754c2fdbe0ba98c1461df4266a2df/packages/gatsby-transformer-react-docgen/src/__tests__/displayname-handler.js#L34

Every entry like this should be changed to

```js
class MyComponent extends React.Component { static displayName = 'foo'; render() {} }
```

**Comment by muescha** [comment]
thanks for the guide :)

i start to work on a PR

**Comment by muescha** [comment]
@vladar 

i created an PR:

- #24788 `fix(plugin): update react-docgen, fix tests`

**Comment by vladar** [comment]
Fixed via #24788
